### DISCUSSION TITLE
CNC and laser in one machine

### DISCUSSION THREAD

**Discussion by axg20202** [phil-barrett/grblHAL-teensy-4.x#54] [Ideas]
Hi Phil.  I'm now mostly up and running with your board and IOsender on my PrintNC.  I see from posts on Lightburn forums that you have had success running a laser with your board. I am wondering if it's possible to rig up a (diode) laser on my CNC and use Lightburn without having to do any rewiring (just power source and PWM for the laser) so that my machine can do double duty simply by connecting up the laser when needed?

**Comment by phil-barrett** [comment]
Yes, it is possible to do. You will need a way to switch your control signal(s) between the spindle and laser.  You will also need to switch grblHAL between normal mode and laser mode ($32).

**Comment by axg20202** [comment]
Ok, great. I’m not currently using the pwm output for my cnc, just the 0-10v for speed control of a vfd. If my vfd is switched off, I could potentially leave those connections in place and simply connect the laser power and pmw leads to the laser when needed, unplugging these at the laser end when notin use, or am I missing something?

**Reply by phil-barrett** [reply]
That will probably work but the devil is in the details. I recommend you have a single interlock switch that controls both the laser and spindle.  Switched one way, the laser works and spindle doesn't, switched the other way, the opposite. Maybe a DPDT connecting the right signals. Having multiple switches or connectors to control which runs is a recipe for error. Don't ask me how I know this!
